samples/sample_hevc_fei/src/pipeline_hevc_fei.cpp

```diff
@@ -320,25 +311,52 @@ mfxStatus CEncodingPipeline::Execute()
     mfxFrameSurface1* pSurf = NULL; // points to frame being processed
 
     mfxU32 numSubmitted = 0;
+    bool insertIDR = false;
+
+    time_t start = time(0);
 
     while (MFX_ERR_NONE <= sts || MFX_ERR_MORE_DATA == sts)
     {
-        if (m_inParams.nNumFrames <= numSubmitted) // frame encoding limit
+        if (m_inParams.nNumFrames <= numSubmitted                               // frame encoding limit
+            || (m_inParams.nTimeout && time(0) - start >= m_inParams.nTimeout)) // encoding time limit
             break;
 
         sts = m_pYUVSource->GetFrame(pSurf);
+        if (MFX_ERR_MORE_DATA == sts && m_inParams.nTimeout && time(0) - start < m_inParams.nTimeout)
+        {
+            // reached end of input file, drain encoder and start from the beginning
```

Comment:
that won't work. You shouldn't drain frames in case of loop encoding. Check presets for B frames, for AVC codec you would see that encoded sequence after draining follows some other preset. The right thing to do here is ONLY reset file pointers to emulate long file, other components should think that they process the next frame from stream

Reply:
Forcing encoder to insert IDR is also required, of course

Reply:
Ok. I'll check that reorder class will work correctly without draining and fix.